kapitan/inputs/helm.py

```diff
@@ -0,0 +1,129 @@
+
+    def compile_file(self, file_path, compile_path, ext_vars, **kwargs):
+        """
+        Render templates in file_path/templates and write to compile_path.
+        file_path must be a directory containing helm chart.
+        kwargs:
+            reveal: default False, set to reveal refs on compile
+            target_name: default None, set to current target being compiled
+        """
+        reveal = kwargs.get("reveal", False)
+        target_name = kwargs.get("target_name", None)
+
+        temp_dir = tempfile.mkdtemp()
+        os.makedirs(os.path.dirname(compile_path), exist_ok=True)
+        # save the template output to temp dir first
+        error_message = self.render_chart(
+            chart_dir=file_path,
+            output_path=temp_dir,
+            helm_values_file=self.helm_values_file,
+            helm_values_files=self.helm_values_files,
+            **self.helm_params,
+        )
+        if error_message:
+            raise HelmTemplateError(error_message)
+
+        walk_root_files = os.walk(temp_dir)
+        for current_dir, _, files in walk_root_files:
+            for file in files:  # go through all the template files
+                rel_dir = os.path.relpath(current_dir, temp_dir)
+                rel_file_name = os.path.join(rel_dir, file)
+                full_file_name = os.path.join(current_dir, file)
+                with open(full_file_name, "r") as f:
+                    item_path = os.path.join(compile_path, rel_file_name)
+                    os.makedirs(os.path.dirname(item_path), exist_ok=True)
+                    with CompiledFile(
+                        item_path,
+                        self.ref_controller,
+                        mode="w",
+                        reveal=reveal,
+                        target_name=target_name,
+                    ) as fp:
+                        yml_obj = list(yaml.safe_load_all(f))
+                        fp.write_yaml(yml_obj)
+                        logger.debug("Wrote file %s to %s", full_file_name, item_path)
+
+        self.helm_values_file = None  # reset this
+        self.helm_params = {}
+        self.helm_values_files = []
+
+    def default_output_type(self):
+        return None
+
+    def render_chart(self, chart_dir, output_path, **kwargs):
+        args = ["template"]
+        """renders helm chart located at chart_dir, and stores the output to output_path"""
+        if kwargs.get("helm_values_files", []):
+            for file_name in kwargs["helm_values_files"]:
+                args.append("-f")
+                args.append(file_name)
+
+        if kwargs.get("helm_values_file", None):
+            args.append("-f")
+            args.append(kwargs["helm_values_file"])
+
+        if kwargs.get("namespace", None):
+            args.append("-n")
+            args.append(kwargs["namespace"])
+
+        args.append("--output-dir")
+        args.append(output_path)
+
+        if self.kube_version:
```

Comment:
This is an amazing PR! I would recommend however to take `helm_params` as an input for all the additional flags like `api-versions`, `validate`, etc. I think it makes it a little more flexible.

Reply:
I pushed a new version that supports most helm flags. There is just a bunch of flags that are disabled because they either affect the output, or are useless for templating and only relevant on install.

Note that `helm template` has a `--release-name` flag that conflicts with the `release_name` helm_params. This version now recommend to use `name` instead of `release_name`. To supports inventory that keep using `release_name`, the helm input check if the `release_name` value is a  `bool`. If not, it prints a deprecation warning and keep using the `release_name` value as `name`.

Reply:
In regards to the conflict of `name` and `release_name`, I think your solution make sense of printing the warning makes sense.